util/gen_world_data.py

```diff
@@ -0,0 +1,133 @@
+import csv
+import coinach
+import json
+import yaml
+from pathlib import Path
+import os
+import re
+import requests
+import sys
+import argparse
+
+base = "https://xivapi.com/"
+
+# First argument is the API key
+if len(sys.argv) > 1:
+    xivapi_key = sys.argv[1]
+else:
+    xivapi_key = False
+
+
+def xivapi(content, filters={}):
+    """Fetches content columns from XIVAPI"""
+    page = 1
+    url = f"{base}{content}"
+    by_id = False
+
+    # IDs are just part of the url path
+    if "id" in filters:
+        url += "/" + str(filters["id"])
+        by_id = True
+        del filters["id"]
+
+    # Add the key
+    url += "?"
+    if xivapi_key:
+        url += f"key={xivapi_key}"
+
+    # Filters are added onto the URL as a query string
+    if len(filters):
+        for key, value in filters.items():
+            url += "&"
+
+            if type(value) is list:
+                # Collapse lists into comma-seperated strings
+                url += f'{key}={",".join(str(x) for x in value)}'
+            else:
+                url += f"{key}={value}"
+
+    response = requests.get(f"{url}&page={page}").json()
+
+    if not by_id:
+        results = response["Results"]
+    else:
+        # Searches by ID do not have pagination separate from results
+        results = response
+
+    # Loop requests until the page is over
+    while not by_id and response["Pagination"]["Page"] != response["Pagination"]["PageTotal"]:
+        page += 1
+        response = requests.get(f"{url}&page={page}")
+        if response.status_code != 200:
+            print(response.status_code)
+            print(response.headers)
+            print(response.text)
+            exit()
+
+        response = response.json()
+
+        results += response["Results"]
+
+    return results
+
+def world_localize(name):
+    d = {
+        "LaNuoXiYa": "拉诺西亚",
```

Comment:
Why are these special? Where did this mapping come from?

Reply:
The official Chinese server.
For example that you point out, `LaNuoXiYa` was the name in code that FFXIV server returned, and Chinese Version Game Client would translate/convert it as `拉诺西亚`. So that Chinese players could use the localized name refers to different servers. On the other hand, `LaNuoXiYa` was the [Pinyin](https://en.wikipedia.org/wiki/Pinyin) of those Chinese Characters `拉诺西亚`.

> I'm not playing in CN server, I just know Chinese. ^_^

Reply:
hmmm... I don't know if we need this localizing thing, but if we need this, here's Korean server list.
krChocobo - 초코보
krMoogle - 모그리
krCarbuncle - 카벙클
krTonberry - 톤베리

Reply:
> hmmm... I don't know if we need this localizing thing, but if we need this, here's Korean world list.
> 
> ```
> "krChocobo": "초코보",
> "krMoogle": "모그리",
> "krCarbuncle": "카벙클",
> "krTonberry": "톤베리",
> ```
> 
> other world name like 'KrCaitsith' 'KrUnicorn' are back up names, which are not used in real, so there's no translation yet.
> Also, there were world(called "server" in Korean) integration twice, here's machine translated table.
> ![Screenshot 2020-04-28 at 18 23 11](https://user-images.githubusercontent.com/4695371/80470816-7a0fe200-897d-11ea-8136-b0b9836c07e0.jpg)
> Some removed world names seem remained until now in the client data, but I think they won't use those world name.

I localized the world names to Korean in https://github.com/quisquous/cactbot/pull/1388/commits/18e9d96274bbd250a4885119559d3d4ed2538b7a .